src/main/java/com/iexec/sms/admin/AdminService.java

```diff
@@ -35,8 +45,62 @@ public AdminService(@Value("${spring.datasource.url}") String datasourceUrl,
         this.datasourcePassword = datasourcePassword;
     }
 
-    public String createDatabaseBackupFile() {
-        return "createDatabaseBackupFile is not implemented";
+    /**
+     * Creates a backup of the H2 database and saves it to the specified location.
+     *
+     * @param storageLocation The location where the backup file will be saved, must be an existing directory.
+     * @param backupFileName  The name of the backup file.
+     * @return {@code true} if the backup was successful; {@code false} if any error occurs.
+     */
+    boolean createDatabaseBackupFile(String storageLocation, String backupFileName) {
+        // Check for missing or empty storageLocation parameter
+        if (StringUtils.isBlank(storageLocation)) {
+            log.error("storageLocation must not be empty.");
+            return false;
+        }
+        // Check for missing or empty backupFileName parameter
+        if (StringUtils.isBlank(backupFileName)) {
+            log.error("backupFileName must not be empty.");
+            return false;
+        }
+        // Ensure that storageLocation ends with a slash
+        if (!storageLocation.endsWith("/")) {
+            storageLocation += "/";
+        }
+
+        // Check if storageLocation is an existing directory, we don't want to create it.
+        final File directory = new File(storageLocation);
+        if (!directory.isDirectory()) {
+            log.error("storageLocation must be an existing directory [storageLocation:{}]", storageLocation);
+            return false;
+        }
+
+        return databaseDump(storageLocation + backupFileName);
+    }
+
+    /**
+     * @param fullBackupFileName complete fileName (location and filename)
+     * @return {@code true} if the backup was successful; {@code false} if any error occurs.
+     */
+    boolean databaseDump(String fullBackupFileName) {
+
+        if (fullBackupFileName == null || fullBackupFileName.isEmpty()) {
+            log.error("fullBackupFileName must not be empty.");
+            return false;
+        }
+
+        try {
+            log.info("Starting the backup process [fullBackupFileName:{}]", fullBackupFileName);
+            final long start = System.currentTimeMillis();
+            Script.process(datasourceUrl, datasourceUsername, datasourcePassword, fullBackupFileName, "DROP", "");
+            final long stop = System.currentTimeMillis();
+            final long size = (new File(fullBackupFileName)).length();
```

Comment:
I think it should be allowed to write this
```suggestion
            final long size = new File(fullBackupFileName).length();
```

Reply:
done in local